src/client.rs

```diff
@@ -30,32 +31,32 @@ use super::{Request, Response};
 use util::HashableValue;
 use error::Error;
 
+
 /// An interface for a transport over which to use the JSONRPC protocol.
-pub trait Transport {
-    /// The Error type for this transport.
-    /// Errors will get converted into Box<std::error::Error> so the
-    /// type here is not use any further.
-    type Err: ::std::error::Error;
-
-    /// Make an RPC call over the transport.
-    fn call<R>(&self, impl serde::Serialize) -> Result<R, Self::Err>
-        where R: for<'a> serde::de::Deserialize<'a>;
+pub trait Transport: fmt::Debug + 'static {
+    /// Send an RPC request over the transport.
+    fn send_request(&self, Request) -> Result<Response, Error>;
+    /// Send a batch of RPC requests over the transport.
+    fn send_batch(&self, &[Request]) -> Result<Vec<Response>, Error>;
+	/// Format the target of this transport.
+	/// I.e. the URL/socket/...
+	fn fmt_target(&self, f: &mut fmt::Formatter) -> fmt::Result;
 }
 
 /// A JSON-RPC client.
 ///
 /// Create a new Client using one of the transport-specific constructors:
 /// - [Client::simple_http] for the built-in bare-minimum HTTP transport
-pub struct Client<T: Transport> {
-    transport: T,
+pub struct Client {
+    pub(crate) transport: Box<dyn Transport>,
```

Comment:
This made it no longer `Send` :(

Reply:
Oh that's because we need to make `Transport: Send` I think. Box should be Send, I discussed that with Andrew.